hack/kubeadminkubeconfig/get.go

```diff
@@ -64,8 +66,19 @@ func main() {
 		log.Fatalf("usage: %s resourceid\n", os.Args[0])
 	}
 
-	err := writeKubeconfig(ctx, os.Args[1])
-	if err != nil {
-		log.Fatal(err)
-	}
+	// Time out after 10 seconds
+	stopCh := make(chan struct{})
+	go func() {
+		time.Sleep(10 * time.Second)
+		close(stopCh)
+	}()
+
+	// Retry every 0.5s on error until success or above timeout
+	wait.PollImmediateUntil(time.Second/2, func() (bool, error) {
+		if err := writeKubeconfig(ctx, os.Args[1]); err != nil {
+			log.Fatal(err)
+			return false, nil
+		}
+		return true, nil
+	}, stopCh)
```

Comment:
How about this (it's closer to the pattern that we use all over the codebase):
Also "log.Fatal()" call sys.Exit() so you dont' want that in the retry loop

```golang
	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Minute)
	defer cancel()

	err = wait.PollImmediateUntil(2*time.Second, func() (bool, error) {
		if err = writeKubeconfig(ctx, os.Args[1]); err != nil {
			log.Warnf("Retrying on error %v", err)
			return false, nil
		}
		return true, err
	}, timeoutCtx.Done())
	if err != nil {
		log.Fatal(err)
	}
```

Reply:
Updated